libcylinder/src/lib.rs

```diff
@@ -15,4 +16,118 @@
  * ------------------------------------------------------------------------------
  */
 
+mod error;
 pub mod signing;
+
+pub use error::{SignatureVerificationError, SigningError};
+
+/// A public key
+pub struct PublicKey {
+    bytes: Vec<u8>,
+}
+
+impl PublicKey {
+    /// Creates a new public key
+    pub fn new(bytes: Vec<u8>) -> Self {
+        Self { bytes }
+    }
+
+    /// Returns the public key as a hex string
+    pub fn as_hex(&self) -> String {
+        hex::encode(&self.bytes)
```

Comment:
Can you use the local hex functions in this commit instead of adding it here and removing it later. (same for below)

Reply:
Updated